scripts/generate_vulkaninfo.py

```diff
@@ -0,0 +1,843 @@
+                        'phys_device_mem_props2': {'extends': 'VkPhysicalDeviceMemoryProperties2', 'type': 'device'},
+                        'phys_device_features2': {'extends': 'VkPhysicalDeviceFeatures2,VkDeviceCreateInfo', 'type': 'device'},
+                        'surface_capabilities2': {'extends': 'VkSurfaceCapabilities2KHR', 'type': 'both'},
+                        'format_properties2': {'extends': 'VkFormatProperties2', 'type': 'device'}
+                        }
+
+
+class VulkanInfoGeneratorOptions(GeneratorOptions):
+    def __init__(self,
+                 conventions=None,
+                 input=None,
+                 filename=None,
+                 directory='.',
+                 apiname=None,
+                 profile=None,
+                 versions='.*',
+                 emitversions='.*',
+                 defaultExtensions=None,
+                 addExtensions=None,
+                 removeExtensions=None,
+                 emitExtensions=None,
+                 sortProcedure=None,
+                 prefixText="",
+                 genFuncPointers=True,
+                 protectFile=True,
+                 protectFeature=True,
+                 protectProto=None,
+                 protectProtoStr=None,
+                 apicall='',
+                 apientry='',
+                 apientryp='',
+                 indentFuncProto=True,
+                 indentFuncPointer=False,
+                 alignFuncParam=0,
+                 expandEnumerants=True,
+                 ):
+        GeneratorOptions.__init__(self, conventions, filename, directory, apiname, profile,
+                                  versions, emitversions, defaultExtensions,
+                                  addExtensions, removeExtensions, emitExtensions, sortProcedure)
+        self.input = input
+        self.prefixText = prefixText
+        self.genFuncPointers = genFuncPointers
+        self.protectFile = protectFile
+        self.protectFeature = protectFeature
+        self.protectProto = protectProto
+        self.protectProtoStr = protectProtoStr
+        self.apicall = apicall
+        self.apientry = apientry
+        self.apientryp = apientryp
+        self.indentFuncProto = indentFuncProto
+        self.indentFuncPointer = indentFuncPointer
+        self.alignFuncParam = alignFuncParam
+
+# VulkanInfoGenerator - subclass of OutputGenerator.
+# Generates a vulkan info output helper function
+
+
+class VulkanInfoGenerator(OutputGenerator):
+
+    def __init__(self,
+                 errFile=sys.stderr,
+                 warnFile=sys.stderr,
+                 diagFile=sys.stdout):
+        OutputGenerator.__init__(self, errFile, warnFile, diagFile)
+
+        self.constants = {}
+
+        self.extension_sets = {}
+        for ext_cat in EXTENSION_CATEGORIES.keys():
+            self.extension_sets[ext_cat] = set()
+
+        self.enums = set()
```

Comment:
Sets and dicts have an undefined iteration order which kills us when we try to verify pre-generated source code in CI.  Workaround is to use `sorted()` at output time on sets and use `collections.OrderedDict` for dictionaries.